akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/Running.scala

```diff
@@ -200,6 +243,90 @@ private[akka] object Running {
       }
     }
 
+    def onPublishedEvent(state: Running.RunningState[S], event: PublishedEventImpl): Behavior[InternalProtocol] = {
+      setup.activeActive match {
+        case None =>
+          setup.log
+            .warn("Received published event for [{}] but not an active active actor, dropping", event.persistenceId)
+          this
+
+        case Some(activeActive) =>
+          event.replicaId match {
+            case None =>
+              setup.log.warn("Received published event for [{}] but with no replica id, dropping")
+              this
+            case Some(replicaId) =>
+              onPublishedEvent(state, activeActive, replicaId, event)
+          }
+      }
+    }
+
+    private def onPublishedEvent(
+        state: Running.RunningState[S],
+        activeActive: ActiveActive,
+        originReplicaId: String,
+        event: PublishedEventImpl): Behavior[InternalProtocol] = {
+      val log = setup.log
+      val separatorIndex = event.persistenceId.id.indexOf(PersistenceId.DefaultSeparator)
+      val idPrefix = event.persistenceId.id.substring(0, separatorIndex)
+      if (!setup.persistenceId.id.startsWith(idPrefix)) {
+        log.warn("Ignoring published replicated event for the wrong actor [{}]", event.persistenceId)
+        this
+      } else if (originReplicaId == activeActive.replicaId) {
+        // FIXME MultiDC handled this differently not sure why we'd need our own published events?
+        if (log.isDebugEnabled)
+          log.debug(
+            "Ignoring published replicated event with seqNr [{}] from our own replica id [{}]",
+            event.sequenceNumber,
+            originReplicaId)
+        this
+      } else if (!activeActive.allReplicas.contains(originReplicaId)) {
+        log.warnN(
+          "Received published replicated event from replica [{}], which is unknown. Active active must be set up with a list of all replicas (known are [{}]).",
```

Comment:
All replicas must be the same on all nodes? This could happen as part of a rolling change to replicas

Reply:
Hmm. Would that be safe for the other parts of active active you mean?

Reply:
Actually, it only affects the optimization, and when the rolling upgrade is complete it will start working (so not a catastophy).

Reply:
Yep i was just thinking out loud. I think dropping them is fine